### PR TITLE
kubeadm: cleanup unnecessary slice initialization statements

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
kubeadm: cleanup unnecessary slice initialization statements.
Named return values do not need to be explicitly initialized.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/preflight/checks_linux.go [hunk 1]
```diff
@@ -29,21 +29,19 @@ import (
 
 // Check validates if Docker is setup to use systemd as the cgroup driver.
 func (idsc IsDockerSystemdCheck) Check() (warnings, errorList []error) {
-	warnings = []error{}
 	driver, err := util.GetCgroupDriverDocker(exec.New())
 	if err != nil {
-		errorList = append(errorList, err)
-		return nil, errorList
+		return nil, append(errorList, err)
```
Comment (draveness) [inline cmd/kubeadm/app/preflight/checks_linux.go]: what do you think of `[]error{err}`

File: cmd/kubeadm/app/preflight/checks_linux.go [hunk 2]
```diff
@@ -29,21 +29,19 @@ import (
...
 	}
 	if driver != util.CgroupDriverSystemd {
 		err = errors.Errorf("detected %q as the Docker cgroup driver. "+
 			"The recommended driver is %q. "+
 			"Please follow the guide at https://kubernetes.io/docs/setup/cri/",
 			driver,
 			util.CgroupDriverSystemd)
-		warnings = append(warnings, err)
+		return append(warnings, err), nil
```
Comment (draveness) [inline cmd/kubeadm/app/preflight/checks_linux.go]: `[]error{err}, nil`

**Review by draveness — CHANGES_REQUESTED** [review]

File: cmd/kubeadm/app/preflight/checks_linux.go (hunk 1 above)
Comment (SataQiu) [reply cmd/kubeadm/app/preflight/checks_linux.go]: Thanks @draveness 
In fact, I did think about it. 
Using `append` can keep consistent with the code base, but `[]error{err}` maybe more efficient.
Simple Test:
```go
func M1() (errs []error) {
	return append(errs, fmt.Errorf("Err"))
}

func M2() (errs []error) {
	return []error{fmt.Errorf("Err")}
}

func Benchmark_M1(b *testing.B) {
	for i := 0; i < b.N; i++ {
		M1()
	}
}

func Benchmark_M2(b *testing.B) {
	for i := 0; i < b.N; i++ {
		M2()
	}
}
```
```sh
$ go test -test.bench=".*"
Benchmark_M1-4          10000000               210 ns/op
Benchmark_M2-4          10000000               199 ns/op
```
WDYT? @neolit123

Comment (dixudx) [reply cmd/kubeadm/app/preflight/checks_linux.go]: `append` will move the array to a new memory address. That's why it will be a little bit slower.

Comment (dixudx) [reply cmd/kubeadm/app/preflight/checks_linux.go]: `(warnings, errorList []error)` currently we have named variables here.

I'd suggest we only need to change to `([]error, []error)` and use `[]error{xxErr}`.

Comment (draveness) [reply cmd/kubeadm/app/preflight/checks_linux.go]: > Using `append` can keep consistent with the code base, but `[]error{err}` maybe more efficient.

Please take a look at https://github.com/kubernetes/kubernetes/pull/81275/files#diff-a466e16a4b5d497feb0dc3539119ed25R173 which use `[]error{err}`

**Review by draveness — CHANGES_REQUESTED** [review]
IMO we need to refactor the code in these files, many of the `errorList` and append operations in these files are not necessary

File: cmd/kubeadm/app/preflight/checks_linux.go (hunk 1 above)
Comment (draveness) [reply cmd/kubeadm/app/preflight/checks_linux.go]: > `(warnings, errorList []error)` currently we have named variables here.
> 
> I'd suggest we only need to change to `([]error, []error)` and use `[]error{xxErr}`.

have a named variable is useful for understanding, `([]error, []error)` could be confused.

Comment (SataQiu) [reply cmd/kubeadm/app/preflight/checks_linux.go]: All right. I will change it later. Thanks @dixudx and @draveness

**Comment by SataQiu** [comment]
> 
> 
> IMO we need to refactor the code in these files, many of the `errorList` and append operations in these files are not necessary

Yes! That's what I'm trying to say. We're using it in a confusing way.

**Review by timothysc — COMMENTED** [review]
I'm kind of 0+ on this.  
/assign @neolit123

**Comment by neolit123** [comment]
@dixudx and @draveness
please, let me know if this looks good to you.

**Review by dixudx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**